lib/telephone/service.rb

```diff
@@ -16,6 +16,18 @@ class Service
     #   Telephone::Service.call(foo: bar).result # "baz"
     attr_accessor :result
 
+    ##
+    # Primary responsibility of initialize is to instantiate the
+    # attributes of the service object with the expected values.
+    def initialize(attributes = {})
```

Comment:
Not sure if you have any thoughts about this, but the following might be unexpected behavior:

```ruby
class SomeService < Telephone::Service
  argument :foo, required: true
  
  def call
    puts "foo: #{foo}"
  end
end

service = SomeService.new(foo: nil)

# What happens 👇
service.call
#=> "foo: "

# What I'd expect since, `foo` is required 👇
service.call
#=> nil
```

In other words, if you run `#call` directly from an instance, you aren't running this line either

```ruby
instance.result = instance.call if instance.valid?
```

Reply:
Interesting callout.

So now, `#call` and `self.call` always return the instance of the object.
```ruby
service = SomeService.new(foo: nil)

service.call
=> service
service.result
=> nil
service.valid?
=> false
service.result
=> nil
```